tests/backup/backup_test.go

```diff
@@ -1828,186 +1828,184 @@ func killStork() {
 	Expect(err).NotTo(HaveOccurred())
 }
 
-// This test crashes volume driver (PX) while backup is in progress
-var _ = Describe("{BackupCrashVolDriver}", func() {
+
+// This test restart volume driver (PX) while backup is in progress
+var _ = Describe("{BackupRestartPX}", func() {
+	var (
+		appList = Inst().AppList
+	)
+	var preRuleNameList []string
+	var postRuleNameList []string
 	var contexts []*scheduler.Context
-	var namespaceMapping map[string]string
-	taskNamePrefix := "backupcrashvoldriver"
 	labelSelectors := make(map[string]string)
-	volumeParams := make(map[string]map[string]string)
-	bkpNamespaces := make([]string, 0)
-
-	BeforeEach(func() {
-		wantAllAfterSuiteActions = false
-		StartTorpedoTest("BackupCrashVolDriver", "crashes volume driver (PX) while backup is in progress", nil, 0)
-	})
-
-	It("has to complete backup and restore", func() {
-		// Set cluster context to cluster where torpedo is running
-		SetClusterContext("")
-		SetupBackup(taskNamePrefix)
-
-		sourceClusterConfigPath, err := GetSourceClusterConfigPath()
-		Expect(err).NotTo(HaveOccurred(),
-			fmt.Sprintf("Failed to get kubeconfig path for source cluster. Error: [%v]", err))
-
-		SetClusterContext(sourceClusterConfigPath)
-
-		Step("Deploy applications", func() {
-			contexts = make([]*scheduler.Context, 0)
-			for i := 0; i < Inst().GlobalScaleFactor; i++ {
-				taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
-				appContexts := ScheduleApplications(taskName)
-				contexts = append(contexts, appContexts...)
-
-				for _, ctx := range appContexts {
-					// Override default App readiness time out of 5 mins with 10 mins
-					ctx.ReadinessTimeout = appReadinessTimeout
-					namespace := GetAppNamespace(ctx, taskName)
-					bkpNamespaces = append(bkpNamespaces, namespace)
+	CloudCredUIDMap := make(map[string]string)
+	var appContexts []*scheduler.Context
+	var backupLocation string
+	var backupLocationUID string
+	var bkpNamespaces []string
+	var backupName string
+	var clusterUid string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	bkpNamespaces = make([]string, 0)
+	ctx, _ := backup.GetAdminCtxFromSecret()
+	providers := getProviders()
```

Comment:
can you move this inside just before each, avoid function calls outside before each and It

Reply:
Done